prime-router/docs/proposals/0004-git-flow-and-versioning.md

```diff
@@ -0,0 +1,63 @@
+# Background
+
+To date, the PRIME Data Hub has used the following flow for code undergoing development, testing, and production:
+
+> `feature_branch(es)` --> `master` --> `production`
+
+While this flow has served its purpose for the testing and validation of software changes before deployment to production, a simpler workflow that allows for the versioning of the platform is a necessary next step in its maturation.
+
+# Goals
+
+The goal of this proposal is to 1) simplify the flow of code from development to production and 2) to employ versioning for a clear lineage and changelog throughout the platform's lifecycle.
+
+# Proposal
+
+## Versioning
+
+Versioning of the PRIME Data Hub platform can be discerned in two ways - this proposal will focus on the first point:
+  - The overall version of the platform
+  - The version of the platform's constituent components
+
+The overall platform should be versioned in a temporally distinct way, yet allowing for multiple production releases within a single calendar day.  The proposed format is as follows:
+
+> YYYY.0M.0D_MICRO
+
+|YYYY|0M|0D|MICRO
+|---|---|---|---|
+|Full year|Zero-padded month|Zero-padded day|Release number|
+|2019, 2020, ...|01, 10, ...|01, 30, ...|0, 1, 2, ...
+
+The proposed versioning scheme is based off of `Calendar Versioning`<sup>1</sup>.  A separate proposal may leverage `Semantic Versioning`<sup>2</sup> to add granularity to the individual components within the PRIME Data Hub platform.
```

Comment:
So you're proposing deployments use calendar versioning, but we update our individual components to use semantic versioning? I like that hybrid approach.

Reply:
CalVersioning is simple to understand and avoids is this release a major or minor release.

Reply:
Agreed. I'm all for CalVersioning for deployments.